capi-matrix/README.md

```diff
@@ -0,0 +1,78 @@
+# Cluster API Kubernetes/OS version management
+
+## Introduction
+
+We have a cluster-PROVIDER app for every provider that we support (it is our interface to define the cluster structure and configuration). It has a set of templates that allow our customers to create a cluster app with all necessary resources required by a Cluster API implementation. 
+
+In the values of these charts, among all possible parameters we define the Kubernetes version of the cluster and the machine image (AMI) that will be used to run the VMs. These images contain the Kubernetes components with a specific version. There are usually three AMIs defined in every provider: bastion, control-plane and node pools.
+
+This RFC born out of necessity to automate the cluster upgrades. For that reason, team Rocket has discussed and found a solution adding a Kubernetes and OS version in each chart release. Today AWS, GCP anf Open Stack providers pinned the Kubernetes/OS version in the chart. Here we propose an extension to give more flexibility and control to Giant Swarm and the customers.
+
+## User stories
+
+- As a customer, I don't want to worry about the kubernetes version or OS version used by my cluster machines. Just rely on the Giant Swarm default and tested version.
+
+- As a customer, I would like to choose the kubernetes version of my cluster without think which OS AMI correspond to that version.
+
+- As Giant Swarm staff, I want to have the capacity to push new Kubernetes/OS versions and trigger the upgrade automation.
+
+- As Giant Swarm staff, I would like to forbid a specific Kubernetes/OS version to be used in case of security constraints.
+
+- As Giant Swarm staff, I would like to release a new OS version with the same Kubernetes version and roll all customer clusters when a security breach is detected.
+
+## Solution
+
+The solution proposed is to create a cluster-PROVIDER chart release for every Kubernetes/OS version is released. That way will help our automation to roll upgrades automatically when new version is published.
+
+But we would like to have a list of images attached to every Kubernetes version like:
+
+```yaml
+kubernetesVersionImages:
```

Comment:
dumb question: are these names provider specific? or would we have the same names across providers?

Reply:
IIRC the names are produced by our image builder CI for each OS/k8s version published. Am I right @kopiczko ?

Reply:
Correct, the names are chosen by us.